### ISSUE TITLE
Support authenication for plugin repositories 

### ISSUE THREAD

**Issue by jb-2020** [gradle/gradle-build-action#998]
In https://github.com/gradle/gradle-build-action/issues/933 the gradle configurations were enhanced to support a custom plugin repository provided by the user.  These plugin repositories can require authentication but there is currently no mechanism to also provide a credentials block to this configuration. 

It would be helpful if we could also provide a username and password to the custom plugin repository e.g.
```
  repositories {
    maven { 
      url "<private gradle plugin repository>
      credentials {
        username = System.getenv("USERNAME")
        password = System.getenv("PASSWORD")
      }
    }
  }
```

**Comment by bigdaz** [comment]
Yep, this makes sense. Would you be willing to develop, test and submit a fix? 
The implementation should be a pretty simple change to [this file](https://github.com/gradle/gradle-build-action/blob/main/src/resources/init-scripts/gradle-build-action.github-dependency-graph-gradle-plugin-apply.groovy), but I don't have an environment to easily test this functionality.

**Comment by jb-2020** [comment]
@bigdaz I did attempt this a while back and I couldn't seem to use the the `project` credentials:  e.g.

```
    val user: String by project
    val password: String by project
```

I think I'm misunderstanding the startup behavior of gradle and these 'init' scripts, the behavior is different than our typical method of gradle configuration.  The only work around I found was to reference environment variables directly.

**Comment by bigdaz** [comment]
You won't be able to access the project instance from within the init-script. The script is loaded _very_ early well before any project is loaded.

Just a thought: before trying to modify the init-script to load 2 new environment variables, can you try using a repository URL like:
`https://user:password@repo.mycompany.com`? 
This is a valid URL format, but I'm not certain if Gradle will translate it to the correct AUTH headers.
